src/bin/rustup-init.rs

```diff
@@ -163,3 +166,19 @@ fn do_recursion_guard() -> Result<()> {
 
     Ok(())
 }
+
+/// Windows pre-main security mitigations.
+///
+/// This is attempts to defend against malicious DLLs that may sit alongside
+/// rustup-init in the user's download folder.
+#[cfg(windows)]
+pub fn pre_rustup_main_init() {
+    use winapi::um::libloaderapi::{SetDefaultDllDirectories, LOAD_LIBRARY_SEARCH_SYSTEM32};
```

Comment:
Maybe orthogonal, but should we start using windows-sys here?

Reply:
I'm surprised there's not an issue for that already but I do think that's beyond the scope of this PR.